modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -462,6 +462,16 @@ CV__DNN_INLINE_NS_BEGIN
          */
         int addLayerToPrev(const String &name, const String &type, LayerParams &params);
 
+        /**
+         * @brief Replaces the type and parameters of existing layer
+         * @param name   name of the replacing layer.
+         * @param type   new typename of the layer (type must be registered in LayerRegister).
+         * @param params new parameters which will be used to initialize the creating layer.
+         *
+         * Intended for internal use only in conjunction with DNN_DIAGNOSTIC_RUN.
+         */
+        void replaceLayer(const String &name, const String &type, LayerParams &params);
```

Comment:
IMHO, we don't need this in public API (User visible). The same comment is about `DummyLayer`.
use `dnn/src/dnn_common.hpp` instead.

Reply:
Moved `NotImplemented` layer to `dnn_common.hpp`. `replaceLayer` became a part of `addLayer` under `if (DNN_DIAGNOSTICS_RUN && type == "NotImplemented")` which will otherwise throw an exception as before.